src/Network/Wai/SAML2/Validation.hs

```diff
@@ -211,14 +214,13 @@ validateSAMLResponse cfg responseXmlDoc samlResponse now = do
     -- all checks out, return the assertion
     pure assertion
 
-decryptAssertion :: SAML2Config -> EncryptedAssertion -> ExceptT SAML2Error IO Assertion
-decryptAssertion cfg encryptedAssertion = do
-    --  ***ASSERTION DECRYPTION***
-    -- the SAML assertion is AES-encrypted and we need to acquire the key
-    -- to decrypt it; the key itself is RSA-encrypted:
-    -- get the private key from the configuration and use it to decrypt
-    -- the key used to decrypt the assertion
-    let pk = saml2PrivateKey cfg
+--  ***ASSERTION DECRYPTION***
+-- the SAML assertion is AES-encrypted and we need to acquire the key
+-- to decrypt it; the key itself is RSA-encrypted:
+-- get the private key from the configuration and use it to decrypt
+-- the key used to decrypt the assertion
+decryptAssertion :: PrivateKey -> EncryptedAssertion -> ExceptT SAML2Error IO Assertion
+decryptAssertion pk encryptedAssertion = do
```

Comment:
I don't think moving this comment makes much sense, although it would be good to have a Haddock comment for `decryptAssertion`, which explains how the different parameters are used etc. -- if it then turns out that the comment in the body is redundant, we can remove it then.

Reply:
I think this function is pretty self-explanatory along with its type signature. I simply removed the comment; I hope it's fine because this isn't exported either

Reply:
I like to have Haddock comments for non-exported definitions as well (see contributing guidelines), even if it is something simple as 
```haskell
-- | `decryptAssertion` @key encryptedAssertion@ decrypts the RSA-encrypted @encryptedAssertion@ using the corresponding private @key@.
```

Reply:
I added a corrected version of your suggestion

Reply:
Thank you! Your comment is definitely more accurate.